src/style.rs

```diff
@@ -674,12 +687,99 @@ enum Alignment {
     Right,
 }
 
+/// Trait for defining stateful or stateless formatters
```

Comment:
I don't think we should elaborate on "stateful" vs "stateless" here, this trait simply facilitates both. People can see the impl for `Fn(&ProgressState) -> String` so should be clear that stateless is an option. Instead, should talk more about the core idea this enables, keeping your own state based on the basic `ProgressState` and generating output from that.

Maybe also call out how `default()` is used.

Reply:
Yeah, renaming it is good!